cmds/webboot/webboot.go

```diff
@@ -0,0 +1,307 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/u-root/webboot/pkg/dhclient"
+	"github.com/u-root/webboot/pkg/menus"
+)
+
+const (
+	tcURL = "http://tinycorelinux.net/10.x/x86_64/release/TinyCorePure64-10.1.iso"
+)
+
+//DownloadDistro contains information of downloadable isos.
+type DownloadDistro struct {
+	URL string
+}
+
+// bookmark record the list of iso webboot allow user to download,
+// and users can enter these label to indicate an iso to download.
+var bookmark = map[string]*DownloadDistro{
+	"tinycore": &DownloadDistro{
+		tcURL,
+	},
+}
+
+// kernelPath is for hardcode the path to kernel. should be replaced by u-root later
+var kernelPath = map[string]string{
+	"archlinux-2020.06.01-x86_64.iso": "arch/boot/x86_64/vmlinuz",
+	"TinyCore-11.1.iso":               "boot/vmlinuz",
+}
+
+var (
+	isoDir  = flag.String("dir", "", "set the iso directory path")
+	timeout = flag.Int("timeout", 15, "Lease timeout in seconds")
+	retry   = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
+	v       = flag.Bool("verbose", false, "Verbose output")
+	ipv4    = flag.Bool("ipv4", true, "use IPV4")
+	ipv6    = flag.Bool("ipv6", true, "use IPV6")
+
+	verbose = func(string, ...interface{}) {}
+)
+
+// ISO type contains information of a iso file, incluing its name,
+// its filepath, the path to its kernel and if it should be choose by default.
+type ISO struct {
+	name      string
+	path      string
+	kernel    string
+	isDefault bool
+}
+
+// DownloadOption contains information of the iso user want to download,
+// include it's name and the download URL
+type DownloadOption struct {
+	isoName string
+	url     string
+}
+
+// DownloadByLinkOption contains information of the iso user want to download,
+// include it's name and the download URL
+type DownloadByLinkOption struct {
+	isoName string
+	url     string
+}
+
```

Comment:
downloadoption and downloadbylinkoption are two nearly identical structs. If you created an interface called BootMedia, it could have several methods, 
type BootMedia interface {
IsDefault() bool
Name() string
Path() string // This can be a URL, a directory, or whatever
Kernel() string
}

This is just an idea. See what you think.